api/src/main/java/org/pmiops/workbench/cohorts/CohortService.java

```diff
@@ -68,11 +89,36 @@ public CohortService(
         .build();
   }
 
-  public TerraJob submitGenomicsCohortExtractionJob(String workspaceNamespace, String workspaceName)
+  private List<String> getPersonIds(Long cohortId) {
+    String cohortDefinition = cohortDao.findOne(cohortId).getCriteria();
+
+    final SearchRequest searchRequest = new Gson().fromJson(cohortDefinition, SearchRequest.class);
+
+    final QueryJobConfiguration participantIdQuery =
+        cohortQueryBuilder.buildParticipantIdQuery(new ParticipantCriteria(searchRequest));
+
+    return Streams.stream(
+            bigQueryService
+                .executeQuery(bigQueryService.filterBigQueryConfig(participantIdQuery))
+                .getValues())
+        .map(personId -> personId.get(0).getValue().toString())
+        .collect(Collectors.toList());
+  }
+
+  public TerraJob submitGenomicsCohortExtractionJob(DbWorkspace workspace, Long cohortId)
       throws ApiException {
     WorkbenchConfig.WgsCohortExtractionConfig cohortExtractionConfig =
         workbenchConfigProvider.get().wgsCohortExtraction;
 
+    FirecloudWorkspace fcWorkspace = fireCloudService.getWorkspace(workspace).get().getWorkspace();
+    String extractionUuid = UUID.randomUUID().toString();
+    String filename = extractionUuid + "_person_ids.txt";
+    Blob personIdsFile =
+        cloudStorageService.writeFile(
+            fcWorkspace.getBucketName(),
+            filename,
+            String.join("\n", getPersonIds(cohortId)).getBytes(StandardCharsets.UTF_8));
```

Comment:
I'm concerned this is too slow, but this could be a follow-up ticket to avoid starting with a high complexity solution. Anecdotally, how large of cohorts have you tested on and what kind of timing do you see?

Reply:
Ah, I had a PR comment about this but it looks like it got "Outdated" with a clean up PR.

I tested up to 100k and it was pretty fast which is why I opted to just roll with this for now.